ironfist/src/cpp/game/save.cpp

```diff
@@ -235,7 +235,7 @@ void SaveMapVariables(ironfist_map::map_t& m) {
 	GetNextMapVariable(key, mapVariableId, mapVariableValue);
 	while (mapVariableId != "noMapVariables") {
 		if (mapVariableValue == NULL) {
-			char errorMessage[50];
+			char *errorMessage = (char *)malloc(strlen(mapVariableId) + 37);

```

Comment:
One of the big concepts in software engineering is coupling.  How much does one thing depend on another? If thing A changes, in what ways will thing B have to change?

Here, thing A are your string constants "MapVariable: " and "'could not be saved.", and thing B is the number 37. 

Imagine: You are an Ironfist player who knows some coding, and, for whatever reason, you don't like the "MapVariable cannot be saved" popup. Maybe you want to add some more information, or maybe you want to translate it? You download the codebase, search for the string "could not be saved", change it to something else.....and never notice 3 lines above there's that number 37 which needs to be changed as well. Oops.

The easy part of software engineering is getting things working. The hard part is making it hard to accidentally break things.

What's something you can do to eliminate this coupling? Think about it for a few minutes before scrolling down. I will post one possible answer in the next comment

---------------

In other news, this is a memory leak: errorMessage is allocated but never released. If you do go this route, you'll need to call free(errorMessage) when you're done using it. Actually, you should probably use the ALLOC and FREE macros rather than malloc and free, to stay consistent with the rest of the codebase. (ALLOC does some additional tracking of allocated memory to help debug memory leaks.)